src/datatable.py

```diff
@@ -0,0 +1,192 @@
+from collections import namedtuple
+from src import mongo
+from querystring_parser import parser
+import json
+
+max_string_length = 50
+
+
+class DataTablesServer:
+
+    def __init__(self, request, columns, index, collection, group_by=None):
+        self.columns = columns
+        self.index = index
+        self.collection = collection
+        self.group_by = group_by
+
+        # values specified by the datatable for filtering, sorting, paging
+        self.request = parser.parse(request.query_string)
+        # print(json.dumps(self.request, sort_keys=True, indent=4))
+
+        self.dbh = mongo.db
+        self.result_data = None  # results from the db
+        self.records_filtered = 0  # total in the table after filtering
+        self.records_total = 0  # total in the table unfiltered
+
+        self.run_queries()
+
+    def output_result_on_given_fields(self):
+        output = {}
+        output['draw'] = str(int(self.request['draw']))
+        output['recordsTotal'] = str(self.records_total)
+        output['recordsFiltered'] = str(self.records_filtered)
+
+        data_rows = []
+        i = self.paging().start
+
+        for row in self.result_data:
+            data_row = []
+            i += 1
+            data_row.append("<a href='/vus/{0}'>{1}</a>".format(row['_id'], i))
+            for col in self.columns:
+                if col in row:
+                    val = row[col]
+                else:
+                    val = "-"
+
+                if isinstance(val, dict):
+                    print("YAS")
+                    val = json.dumps(val)
+                    print(val)
+                elif isinstance(val, list):
+                    val = ", ".join(val)
+
+                if isinstance(val, str):
+                    val = (val[:max_string_length] + '...') if len(val) > max_string_length else val
+
+                uwu = ""
+                if col == "fullgnomen":
+                    uwu = '<a href="/variant/{0}">{0}</a>'.format(val)
+                elif col == "dn_no":
+                    uwu = '<a href="/patient/{0}">{0}</a>'.format(val)
+                elif col == "gene":
+                    uwu = '<a href="/gene/{0}">{0}</a>'.format(val)
+                else:
+                    uwu = val
+                data_row.append(uwu)
+            data_rows.append(data_row)
+        output['data'] = data_rows
+        return output
+
+    def output_result_on_queried_fields(self):
```

Comment:
see earlier comment:
I have made another issue #19, so that the next version needs to have a look.

Reply:
Second question regarding return value of the function is personal preference.

The first question whether `output_result_on_given_fields` and `output_result_on_queried_fields` are code duplications is still valid. Let me know if you agree that it is a code duplication.

Reply:
I was busy with finding out what the difference was. The functions seem to have a lot in common, but they were also very different (if you compared them next to each other).
I tried to find out what each function does exactly. the most used function is "given_fields" and the other does not seem to be used at all. I think I know why.

Before, all text was a link in the VUSualizer. I changed this in my own feature-branches to only gene, variant and patient. The reasoning was that all other links "did not work" properly. They referenced back to the "list_all" page, but did not seem to do anything there. I suspect that the intention was:
1. I click on "chromosome 5" *you can think of any text here
2. go to the "list all" page and show every variant that is also on chromosome 5
Also seems to explain the filtering going on and the name of the function (queried fields). I removed the entire code here and also the code in main.py. The functionality is not used and did not function 100% yey, so if this needs to be implemented in the future it might be better to start from scratch.